From f5e11b15bc4d746e6e4eb3237170585e470ac1cd Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 21 Mar 2017 10:17:36 +0800
Subject: [PATCH 1584/5242] MLK-14487-2 rpmsg: imx: validate the rpmsg on
 imx7ulp

commit  0b2659775b5d1018cc0f148fb25bd803626e3b69 from
https://source.codeaurora.org/external/imx/linux-imx.git

- add the .data into the imx_rpmsg_dt_ids table, thus
the driver can distinguish the different platforms.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Tested-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/rpmsg/imx_rpmsg.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/rpmsg/imx_rpmsg.c b/drivers/rpmsg/imx_rpmsg.c
index 5d4d301..140c370 100644
--- a/drivers/rpmsg/imx_rpmsg.c
+++ b/drivers/rpmsg/imx_rpmsg.c
@@ -342,9 +342,9 @@ static void imx_rpmsg_vproc_release(struct device *dev)
 };
 
 static const struct of_device_id imx_rpmsg_dt_ids[] = {
-	{ .compatible = "fsl,imx6sx-rpmsg", },
-	{ .compatible = "fsl,imx7d-rpmsg", },
-	{ .compatible = "fsl,imx7ulp-rpmsg", },
+	{ .compatible = "fsl,imx6sx-rpmsg",  .data = (void *)IMX6SX, },
+	{ .compatible = "fsl,imx7d-rpmsg",   .data = (void *)IMX7D, },
+	{ .compatible = "fsl,imx7ulp-rpmsg", .data = (void *)IMX7ULP, },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, imx_rpmsg_dt_ids);
-- 
1.7.9.5

